src/AppBundle/Security/RateLimiter.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace AppBundle\Security;
+
+use AppBundle\Entity\IpRequest;
+use Doctrine\ORM\EntityManager;
+
+class RateLimiter
+{
+    private $em;
+
+    private $ipRequest;
+
+    const AGGREGATION_DELAY = 5;
+    const MAX_ATTEMPTS = 15;
+
+    public function __construct(EntityManager $em)
+    {
+        $this->em = $em;
+    }
+
+    public function getIpRequest(string $ip)
+    {
+        $this->ipRequest = $this->searchOldIpRequest($ip) ?? $this->getNewIpRequest($ip);
+
+        return $this->updateAttempts($this->ipRequest);
+    }
+
+    private function getRateLimitRemaining(): int
+    {
+        $rateLimitRemaining = static::MAX_ATTEMPTS - $this->ipRequest->countAccesses();
+        if ($rateLimitRemaining < 0) {
+            $rateLimitRemaining = 0;
+        }
+
+        return $rateLimitRemaining;
+    }
+
+    private function getRateLimitReset(): int
+    {
+        $dateRequest = clone $this->ipRequest->getDateRequest();
+        $interval = (new \DateTime())->diff($dateRequest);
+
+        $secondBetweenDate = $interval->days * 86400 + $interval->h * 3600
```

Comment:
Comparing timestamps here would be more pertinent. See https://stackoverflow.com/questions/3176609/calculate-total-seconds-in-php-dateinterval